### PR TITLE
Update black version

### PR DESCRIPTION
**Pull request by abhisrkckl**
Update black version from 22.x to 23.x and re-blacken the package.

#1520

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,6 +6,6 @@ repos:
     -   id: check-merge-conflict
     -   id: check-symlinks
 -   repo: https://github.com/psf/black
-    rev: 22.3.0
+    rev: 23.1.0
```
Comment (aarchiba) [inline .pre-commit-config.yaml:9]: According to the black stability policy, any version 23.*.* should be fine, and this allows bug fixes: https://black.readthedocs.io/en/stable/the_black_code_style/index.html#stability-policy

File: requirements_dev.txt [hunk 2]
```diff
@@ -30,7 +30,7 @@ tox
 pre-commit
 typed-ast>=1.5.0
 #black>=19.0a0,<20.0a0
-black>=22.0
+black>=23.0
```
Comment (aarchiba) [inline requirements_dev.txt:33]: This should probably be black 23.*.* - not 24.* or everything will break when that version comes out (again).

File: src/pint/fitter.py [hunk 3]
```diff
@@ -135,7 +135,9 @@ def __get__(self, instance, owner=None):
                 )
             try:
                 cache = instance.__dict__
-            except AttributeError:  # not all objects have __dict__ (e.g. class defines slots)
+            except (
+                AttributeError
+            ):  # not all objects have __dict__ (e.g. class defines slots)
```
Comment (aarchiba) [inline src/pint/fitter.py:140]: Would it help to put the comment on a new line to avoid the ugly bracketing?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (abhisrkckl) [reply .pre-commit-config.yaml:9]: Should this be changed to "rev: 23"?

File: src/pint/fitter.py (hunk 3 above)
Comment (abhisrkckl) [reply src/pint/fitter.py:140]: Yes. I'll do that.

File: tox.ini [hunk 4]
```diff
@@ -125,7 +125,7 @@ skip_install = true
 changedir = .
 description = use black
 deps =
-    black>=22.0
+    black>=23.0
```
[the remark this answers is not in the document]
Comment (abhisrkckl) [reply tox.ini]: changed to black ~= 23.0

File: requirements_dev.txt (hunk 2 above)
Comment (abhisrkckl) [reply requirements_dev.txt]: changed to black ~= 23.0

File: .pre-commit-config.yaml (hunk 1 above)
Comment (abhisrkckl) [reply .pre-commit-config.yaml:9]: It doesn't work without the full version number.

**Comment by aarchiba** [comment]
Does this need a Changelog entry?

It might be helpful to add a section at the end of CONTRIBUTING.rst listing known, regular maintenance/development issues. I'd suggest a note about fixing this every year, and maybe also something about adding new observatories or clock files for existing observatories. Maybe also some notes about how to release a new version - I seem to recall there was some nuance there? Updating the Changelog, perhaps? I seem to recall discussing that with @paulray .

**Comment by abhisrkckl** [comment]
> Does this need a Changelog entry?
Added entry.

> It might be helpful to add a section at the end of CONTRIBUTING.rst listing known, regular maintenance/development issues.
I think this can be a separate PR and it needs some discussion, IMO.

**Pull request merged**